pkg/kube/secrets/manifest_test.go

```diff
@@ -0,0 +1,198 @@
+package secrets_test
+
+import (
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	. "code.cloudfoundry.org/cf-operator/pkg/kube/secrets"
+
+	"code.cloudfoundry.org/cf-operator/pkg/bosh/manifest"
+	yaml "gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	testclient "k8s.io/client-go/kubernetes/fake"
+)
+
+var _ = Describe(`Versioning and persistence for "desired manifests"`, func() {
+	exampleManifest := func(data string) manifest.Manifest {
+		var manifest manifest.Manifest
+		err := yaml.Unmarshal([]byte(data), &manifest)
+		Expect(err).ToNot(HaveOccurred())
+
+		return manifest
+	}
+
+	exampleSourceDescription := "created by a unit-test"
+
+	var namespace = "default"
+	var deploymentName = "fake-deployment"
+
+	Context("Creating a desired manifest secret", func() {
```

Comment:
Interesting. I would have structured the ginkgo unit test it like 

```
Describe 'PersistManifest'
  Context 'when there is no versioned secret'
     BeforeEach (to setup test data)
     It 'creates a secret' (to make the assertion)
Describe 'ListAllVersion'
...
```

Reply:
I think we initially started to arrange the test cases by using the `context` for each list item of the requirements from the tracker card. I am happy to rearrange it based on `describe`.